### ISSUE TITLE
Nodemailer `ECONNREFUSED: Failed to connect`

### ISSUE THREAD

**Issue by SupertigerDev** [oven-sh/bun#6101]
### What version of Bun is running?

1.0.3

### What platform is your computer?

Microsoft Windows NT 10.0.22621.0 x64

### What steps can reproduce the bug?

Keep running the code below, most of the time, it will give a connection error.
```ts
import nodemailer from 'nodemailer';


const transporter = nodemailer.createTransport({
  host: "smtp.gmail.com",
  port: 587,
  auth: {
    user: "username",
    pass: "password",
  },
});

transporter.sendMail({
  from: "someemail@wowie.com",
  to: "someemail@gmail.com",
  subject: "testt",
  html: "test",
  text: "test"
}).then(console.log).catch(console.error)
```

### What is the expected behavior?

It should always always respond with a `Error: Invalid login` error, just like it does in nodejs.
```
fishie@DESKTOP-DPJQI3M:~/test$ node test.js
Error: Invalid login: 454 4.7.0 Too many login attempts, please try again later. k14-20020adfb34e000000b00323330edbc7sm4916796wrd.20 - gsmtp
    at SMTPConnection._formatError (/home/fishie/test/node_modules/nodemailer/lib/smtp-connection/index.js:790:19)
    at SMTPConnection._actionAUTHComplete (/home/fishie/test/node_modules/nodemailer/lib/smtp-connection/index.js:1564:34)
    at SMTPConnection.<anonymous> (/home/fishie/test/node_modules/nodemailer/lib/smtp-connection/index.js:546:26)
    at SMTPConnection._processResponse (/home/fishie/test/node_modules/nodemailer/lib/smtp-connection/index.js:969:20)
    at SMTPConnection._onData (/home/fishie/test/node_modules/nodemailer/lib/smtp-connection/index.js:755:14)
    at SMTPConnection._onSocketData (/home/fishie/test/node_modules/nodemailer/lib/smtp-connection/index.js:193:44)
    at TLSSocket.emit (node:events:514:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Readable.push (node:internal/streams/readable:234:10) {
  code: 'EAUTH',
  response: '454 4.7.0 Too many login attempts, please try again later. k14-20020adfb34e000000b00323330edbc7sm4916796wrd.20 - gsmtp',
  responseCode: 454,
  command: 'AUTH PLAIN'
}
```

### What do you see instead?

It only shows the correct error sometimes, but it mostly shows this error:

```
fishie@DESKTOP-DPJQI3M:~/test$ bun index.ts
218 |     if (this.remotePort = port, socket.data = this, socket.timeout(this.timeout), socket.ref(), this[bunSocketInternal] = socket, this.connecting = !1, !this.#upgraded)
219 |       this.emit("connect", this);
220 |     Socket2.#Drain(socket);
221 |   }
222 |
223 |   connect(port, host, connectListener) {
                     ^
ECONNREFUSED: Failed to connect
   code: "ESOCKET"
 syscall: "connect"

      at connect (node:net:223:18)
      at /home/fishie/test/node_modules/nodemailer/lib/smtp-connection/index.js:354:35
      at /home/fishie/test/node_modules/nodemailer/lib/shared/index.js:152:19
      at /home/fishie/test/node_modules/nodemailer/lib/shared/index.js:66:15
      at node:dns:163:90

fishie@DESKTOP-DPJQI3M:~/test$ ^C
```

### Additional information

• This connects fine in nodejs, but doesnt in bun.
• Even entering correct smtp details, in nodejs, email sends fine, but in bun, it only sends SOMETIMES, but most of the time showing that `ECONNREFUSED` error

**Comment by superbrobenji** [comment]
This doesn't seem to be fixed yet

**Comment by ghost** [comment]
@Jarred-Sumner 

I have the error in nodemailer, too, in actual version `nodemailer@6.9.6`

```batch
ECONNREFUSED: Failed to connect
   code: "ESOCKET"
 syscall: "connect"
```

`bun --revision`
```
1.0.6+969da088f5db3258a803ec186012e30f992829b4
```

`uname -mprs`
```
Linux 5.15.0-86-generic aarch64 aarch64
```

In 7 of 10 ~connection tests.~ When calling `transporter.sendMail(message);`

```js
import nodemailer from "nodemailer";

let transporter = nodemailer.createTransport({
  host: "...",
  port: 587,
  secure: false,
  auth: {
    user: "...",
    pass: "...",
  },
  tls: {
    requireTLS: true,
  }
});

transporter.verify(function (error, success) {
  if (error) {
    console.log(error);
  } else {
    console.log("Server is ready to take our messages");

    let message = {
      from: "...",
      to: "...",
      subject: "Message title",
      text: "Plaintext version of the message",
      html: "<p>HTML version of the message</p>"
    };
    transporter.sendMail(message);

  }
});
```

**Comment by ghost** [comment]
@superbrobenji 

is now fixed for me in `bun 1.0.7` 👍
